docs/deployment-integration.md

```diff
@@ -0,0 +1,150 @@
+
+<!-- TODO: once we have docs for the allow list and rate limiting, we can link
+those here -->
+
+## Rollup BSN Registration
+
+After deploying and instantiating the Rollup BSN contract, the
+rollup must register on Babylon Genesis. Registration requires 
+submitting metadata to identify and describe the BSN
+
+Required metadata for BSN registration:
+* `consumer_id`: BSN ID of your rollup (same as the one you used on contract
+  instantiation)
+* `consumer_name`: Human-readable name of your rollup BSN (e.g., `"DeFi Rollup Chain"`)
+* `consumer_description`: Brief description of the rollup BSN's purpose
+* `max_multi_staked_fps`: Maximum number of rollup FPs per BTC delegation. Prevents delegators from splitting their BTC across too many networks (recommended: `3-5`)
+<!-- TODO: this should not be here, we will have a global limit on fps -->
+* `rollup_finality_contract_address`: Babylon Genesis address of the deployed Rollup BSN contract (`bbn1...` format)
+
+To register the BSN, use:
+```shell
+babylond tx btcstkconsumer register-consumer \
+                                <consumer_id> \
+                                <consumer_name> \
+                                <consumer_ description> \
+                                <max_multi_staked_fps> \
+                                <rollup_finality_contract_address>
+```
+
+> **Note**: The above operation requires governance in permissioned
+> registration networks.
+
+## Contract Maintenance
+
+The Rollup BSN contract exposes `admin-only` functions for ongoing
+contract management. These operations allow storage cleanup, role 
+updates, and future configuration changes without redeployment.
+
+<!-- TODO:
+     1. Add code listings on how the operations are performed.
+     2. Add text describing why someone would modify those.
+        The document should provide understanding on how to be a
+        contract admin, not just on what are the tech capabilities.
+     3. We can also link to the documentations we have (e.g. `PRUNING.md`) -->
+
+### Modifying the Contract Administrator
+
+- `UpdateAdmin`: Transfers contract admin rights to a new Babylon address. Validates the address format and enforces 
+access control using `cw_controllers::Admin`. Executed via `wasm execute` with the `UpdateAdmin` message
+
+### Data Pruning
+
+- `PruneData`: Removes all finality signatures and public randomness for rollup blocks with `height ≤ block_height`. 
+Executed via `wasm execute` with the `PruneData` message. Optional parameters control 
+batch size to prevent gas exhaustion
+
+
+### Modifying the Finality Providers Allow-List
+- `SetFPAllowlist`: Manages the FP allowlist by adding or removing FPs authorized to submit signatures and randomness
+
+### Rate Limiting Submissions 
+- `SetSubmissionInterval`: Updates the submission interval configuration to control how frequently each FP can 
+submit signatures (e.g., every X-th block)
+
+<!-- TODO: what about public randomness parameters? -->
```

Comment:
What public randomness parameters? Currently we only have
`add_to_allowlist`, `remove_from_allowlist` , `update_admin`, `prune_data`

And non admin once
`submit_finality_signature`, `commit_public_randomness`